### PR TITLE
Bug 2111537: (image info): Introduce show-multiarch flag

### PR DESCRIPTION
**Pull request by ardaguclu**
Currently if the passed image is multiarch image, we are throwing error
to force user to use `--filter-by-os` flag with the listed image manifests in
that error.

This PR introduces new `show-multiarch` flag. `oc image info --show-multiarch`
command will continue working exactly the same way as now for single arch images.
However, this command(with new flag) will also show informations of platform specific images
embedded in multiarch image.

This flag supports json and default printing and if the image is multiarch image,
result will be in json array format if output is json.

### CODE REVIEW

File: pkg/cli/image/info/info.go [hunk 1]
```diff
@@ -143,7 +145,7 @@ func (o *InfoOptions) Run() error {
 				icspWarned = true
 			}
 
-			var image *Image
+			var image []*Image
```
Comment (ingvagabund) [inline pkg/cli/image/info/info.go:148]: s/image/images

**Review by ingvagabund — CHANGES_REQUESTED** [review]

File: pkg/cli/image/info/info.go (hunk 1 above)
Comment (ardaguclu) [reply pkg/cli/image/info/info.go]: @ingvagabund thanks. Done.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**